### PR TITLE
Welcome screens updates

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T142778

- [x] Refactor Welcome bits and storyboard to accommodate new layout style.
- [x] Use popover presentation for tablet.
- [x] Merge in develop resolving conflicts.
- [x] Update label text.
- [x] Update animation colors to match mocks.
- [x] Test and tweak all labels to handle longer strings.
- [x] Update "Tell me more" popup text.
- [x] Test on 4s sim.
- [x] Show once even if user has seen the old Welcome screen already.
- [x] RTL - see bottom of [this](https://developer.apple.com/library/content/documentation/WindowsViews/Conceptual/ViewControllerCatalog/Chapters/PageViewControllers.html).

### CODE REVIEW

File: Wikipedia/Code/MWKLanguageLinkController.m [hunk 1]
```diff
@@ -138,13 +138,13 @@ - (void)appendPreferredLanguage:(MWKLanguageLink *)language {
...
 
-- (void)reorderPreferredLanguage:(MWKLanguageLink *)language toIndex:(NSUInteger)newIndex {
+- (void)reorderPreferredLanguage:(MWKLanguageLink *)language toIndex:(NSInteger)newIndex {
     NSMutableArray<NSString *> *langCodes = [[self readPreferredLanguageCodes] mutableCopy];
     NSAssert(newIndex < [langCodes count], @"new language index is out of range");
     if (newIndex >= [langCodes count]) {
         return;
     }
-    NSUInteger oldIndex = [langCodes indexOfObject:language.languageCode];
+    NSInteger oldIndex = [langCodes indexOfObject:language.languageCode];
```
Comment (coreyfloyd) [inline Wikipedia/Code/MWKLanguageLinkController.m]: Objective-c arrays still use NSUInteger - you must cast to be safe

File: Wikipedia/Code/WMFWelcomePageViewController.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            let nextIndex = index! + 1
+            indexForDotIndicator = nextIndex
+            self.setViewControllers([pageControllers[nextIndex]], direction: .Forward, animated: true, completion: {(Bool) in
+                self.view.userInteractionEnabled = true
+            })
+        }
+    }
+
+    private func containerControllerForWelcomePageType(type: WMFWelcomePageType) -> WMFWelcomeContainerViewController {
+        let controller = WMFWelcomeContainerViewController.wmf_viewControllerWithIdentifier("WMFWelcomeContainerViewController", fromStoryboardNamed: "WMFWelcome")
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomePageViewController.swift]: Same thing here… wouldn't expect to need to pass a stringified version of the class name to the method which is a category on the class

File: Wikipedia/Code/WMFWelcomeAnimationView.swift [hunk 3]
```diff
@@ -44,4 +44,23 @@ public class WelcomeAnimationView : UIView {
         // Fix for: http://stackoverflow.com/a/39614714
         self.layoutIfNeeded()
     }
+    
+    public func wmf_beginAnimations() {
+        
+    }
+    
+    public func wmf_addAnimationElementsScaledToCurrentFrameSize(){
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomeAnimationView.swift]: Same thing here

File: Wikipedia/Code/WMFWelcomePanelViewController.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
+
+class WMFWelcomePanelViewController: UIViewController {
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomePanelViewController.swift]: Can you think of a better name for this? Its not really clear what it does from the name

File: Wikipedia/Code/MWKLanguageLinkController.m (hunk 1 above)
Comment (montehurd) [inline Wikipedia/Code/MWKLanguageLinkController.m]: Fixed.

File: Wikipedia/Code/WMFAppViewController.m [hunk 5]
```diff
@@ -797,14 +796,16 @@ - (void)setDidShowOnboarding {
 
 - (void)presentOnboardingIfNeededWithCompletion:(void (^)(BOOL didShowOnboarding))completion {
     if ([self shouldShowOnboarding]) {
-        WMFWelcomeViewController *vc = [WMFWelcomeViewController welcomeViewControllerFromDefaultStoryBoard];
+        WMFWelcomePageViewController *vc = [WMFWelcomePageViewController wmf_viewControllerWithIdentifier:@"WMFWelcomePageViewController" fromStoryboardNamed:@"WMFWelcome"];
```
Comment (montehurd) [inline Wikipedia/Code/WMFAppViewController.m]: Fixed.

File: Wikipedia/Code/WMFWelcomeAnimationView.swift [hunk 6]
```diff
@@ -44,4 +44,23 @@ public class WelcomeAnimationView : UIView {
         // Fix for: http://stackoverflow.com/a/39614714
         self.layoutIfNeeded()
     }
+    
+    public func wmf_beginAnimations() {
```
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnimationView.swift]: Oh crap I was going to try to move WMFWelcomeAnimationView bits into a protocol, but ended up not doing so...

Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnimationView.swift]: Fixed.

File: Wikipedia/Code/WMFWelcomeAnimationView.swift (hunk 3 above)
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnimationView.swift]: Fixed.

File: Wikipedia/Code/WMFWelcomeAnimationView.swift [hunk 7]
```diff
@@ -44,4 +44,23 @@ public class WelcomeAnimationView : UIView {
...
+        
+    }
+    
+    public func wmf_addAnimationElementsScaledToCurrentFrameSize(){
+    
+    }
+    
+    public func wmf_removeExistingSubviewsAndSublayers() {
```
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnimationView.swift]: Fixed.

File: Wikipedia/Code/WMFWelcomeContainerViewController.swift [hunk 8]
```diff
@@ -0,0 +1,52 @@
+
+class WMFWelcomeContainerViewController: WMFWelcomeFadeInAndUpOnceViewController {
```
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeContainerViewController.swift]: Fixed.

**Comment by montehurd** [comment]
@coreyfloyd I think it's good to go. Found a bug with the page control's dots when swiping on RTL iOS 9, but was able to add a hack work-around.

File: Wikipedia/Code/WMFWelcomePanelViewController.swift (hunk 4 above)
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomePanelViewController.swift]: Agreed to leave this for now.

**Pull request merged**